plenum/common/util.py

```diff
@@ -253,13 +255,24 @@ def distributedConnectionMap(names: List[str]) -> OrderedDict:
 
 def checkPortAvailable(ha):
     """Checks whether the given port is available"""
+    def logErrorAndRaise(e, msg=None):
+        erroMsg = msg or "Checked port availability for opening and address, " \
+              "got error: {}".format(str(e))
+        logging.warning(erroMsg)
+        raise e
+
```

Comment:
This does not separate concerns. checkPortAvailable should not be concerned with logging at all. Raise an exception and let application fault barriers handle logging.